addons/website_sale/controllers/main.py

```diff
@@ -200,6 +200,12 @@ def shop(self, page=0, category=None, search='', **post):
 
         product_obj = pool.get('product.template')
 
+        def recursive_category(category, parent_category=[]):
```

Comment:
You should pass `None` and initialize it to [] if None is detected to avoid data sharing.